pennylane/fermi/fermionic.py

```diff
@@ -511,6 +512,69 @@ def from_string(fermi_string):
     return FermiWord({(i, int(s[:-1])): s[-1] for i, s in enumerate(operators)})
 
 
+def to_string(fermi_op, of=False):
+    r"""Return a string representation of the :class:`~.FermiWord` or :class:`~.FermiSentence` object.
+
+    Args:
+        fermi_op (FermiWord, FermiSentence): the fermionic operator
+        of (bool): whether to return a string representation in the same style as OpenFermion using
+                    the shorthand: 'q^' = a^\dagger_q 'q' = a_q. Each operator in the word is represented by the
+                    number of the wire it operates on
+
+    >>> w = FermiWord({(0, 0) : '+', (1, 1) : '-'})
+    >>> to_string(w)
+    0+ 1-
+
+    >>> w = FermiWord({(0, 0) : '+', (1, 1) : '-'})
+    >>> to_string(w, of=True)
+    0^ 1
+
+    >>> w1 = FermiWord({(0, 0) : '+', (1, 1) : '-'})
+    >>> w2 = FermiWord({(0, 1) : '+', (1, 2) : '-'})
+    >>> s = FermiSentence({w1 : 1.2, w2: 3.1})
+    1.2 * 0+ 1-
+    + 3.1 * 1+ 2-
+    """
+    return _to_string_dispatch(fermi_op, of=of)
+
+
+@singledispatch
+def _to_string_dispatch(fermi_op, of=False):
+    """Dispatches to appropriate function if fermi_op is a FermiWord or FermiSentence."""
+    raise ValueError(f"fermi_op must be a FermiWord or FermiSentence, got: {type(fermi_op)}")
+
+
+@_to_string_dispatch.register
+def _(fermi_op: FermiWord, of=False):
+    pl_to_of_map = {"+": "^", "-": ""}
+
+    if len(fermi_op) == 0:
+        return "I"
+
+    op_list = ["" for _ in range(len(fermi_op))]
+    for loc, wire in fermi_op:
+        if of:
+            op_str = str(wire) + pl_to_of_map[fermi_op[(loc, wire)]]
+        else:
+            op_str = str(wire) + fermi_op[(loc, wire)]
+
+        op_list[loc] += op_str
+
+    return " ".join(op_list).rstrip()
+
+
+@_to_string_dispatch.register
+def _(fermi_op: FermiSentence, of=False):
+    if len(fermi_op) == 0:
+        return "I"
```

Comment:
I am not very familiar with Fermi arithmetic so correct me if I am wrong, but in the case of PauliSentence vs. PauliWord: 

- An empty PauliWord represents the multiplicative identity (I)
- An empty PauliSentence represents the additive identity(0)

In this case it seems they both represent the multiplicative identity. Is this correct?

Reply:
Good catch, thank you! The code has been updated accordingly.